barn2/barn2.py

```diff
@@ -0,0 +1,28 @@
+"""
+ID: whuan2001
+LANG: PYTHON2
+TASK: barn1
+"""
+def solution():
+    fin = open("barn1.in", "r")
+    fout = open("barn1.out", "w")
+    max_purchase, total_barns, occupied = map(int, fin.readline().strip().split())
+    if max_purchase >= occupied:
+        print >> fout, occupied
+        return
+    arr = []
+    for gate in fin.readlines():
+        arr.append(int(gate.strip()))
+    arr = sorted(arr)
+    brr = []
+    for i in range(1, len(arr)):
+        brr.append(arr[i] - arr[i - 1])
```

Comment:
You can replace this loop with [succ - curr for succ, curr in zip(arr[1:], arr)]

zip([1, 2, 3], [2, 3, 4]) => [[1, 2], [2, 3], [3, 4]]
zip([1, 2], [2, 3, 4]) => [[1, 2], [2, 3]]
zip([a2, a3, a4], [a1, a2, a3]) => [[a2, a1],